src/webgpu/util/floating_point.ts

```diff
@@ -708,6 +709,39 @@ export abstract class FPTraits {
     return new FPInterval(this.kind, n, n);
   }
 
+  /**
+   * WGSL specifies unbounded precision:
+   *   https://www.w3.org/TR/WGSL/#floating-point-accuracy
+   * In most computations doubles (number in js) can provide enough
+   * precision to give the correctly rounded interval.
+   * However in some cases (addition/subtraction) doubles clearly do
+   * not provide enough numeric precision.
+   * These cases are whenever a large number (eg 1.0) is added or
```

Comment:
nit/suggested rewrite:

These cases are whenever a number, `x` (e.g. 1.0) is added (or subtracted) from a much smaller number, `y` (e.g. smallest positive normal), such that the difference between them is significantly smaller then `ULP(x)`. When working in JS `number`s the result will, incorrectly, simply be `x`, since it will get rounded. We must detect these cases where double precision does not represent infinitely accurate computations accurately then we must manually create the interval.